### PR TITLE
Log pod details

### PR DESCRIPTION
**Pull request by amielkelinkedin**
This change allows the logging of Events and Pod Logs for specifics types of failures to the UI before pod deletion.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1216]: Why is this check required ?

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1196]: Nit: `formatted` (spelling)

File: azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java:321]: Why is this sleep needed ?

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1080]: Note that `deletePods` is expected to be a non-blocking call (or close to one). Adding multiple db queries, and k8s-api calls and  esp any sleep violates that assumption and can create problems for callers of this method as they will indirectly be blocked.
Any thoughts around restructuring things differently ? (happy to discuss offline)

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1081]: Nit: maybe use more descriptive name like `flowStatus` or even `status`.

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1080]: Callers of `deletePod` likely already have access to the `ExecutableFlow` of the object, could you please check if this additional call to `fetchExecutableFlow` can be avoided?

File: .travis.yml
Comment (sshardool) [inline .travis.yml:32]: Why do we need to increase this ?

File: azkaban-common/src/main/java/azkaban/executor/Status.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/Status.java:112]: Perhaps a more generic name would be better suited, something like `isStatusFinishedWithoutSuccess`.

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerizedImpl.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/container/ContainerizedImpl.java:23]: Currently this doesn't seem to be invoked anywhere beyond `KubernetesContainerizedImpl`, is that intentional ?
(we could get away w/o adding it to the interface then)

**Review by sshardool — COMMENTED** [review]
Thanks a lot for the changes, this will be very useful for debugging future issues. 

Please add more detailed description of the PR in the commit message (including a more descriptive heading). 
Haven't reviewed the entire PR yet, noting some comments for early feedback.

File: .travis.yml
Comment (amielkelinkedin) [reply .travis.yml:32]: Our travis build is flaky due to failures from testFlowManagerListenerCreateContainerError. This fixes that problem.

File: azkaban-common/src/main/java/azkaban/executor/Status.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/Status.java:112]: Wonderful suggestion. Will fix.

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerizedImpl.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/container/ContainerizedImpl.java:23]: I think long term we should be logging our PodSpec.yaml in logger.debug mode. This will enable a kubectl apply -f of the yaml which will help in debugging. Not sure where that needs to be added. I can remove it here, and from the interface and the Mock if you think that should be a future consideration.

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1216]: This is the null case. Perhaps there is a better way to do this. Let me think on it.

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1080]: That is a wonderful point. The idea is that we must always delete pods, right? There can't be an exception thrown which would prevent pod deletion and make it so that pods accumulate in the k8s cluster?

File: azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java:70]: Can you add javadoc style comment on top of the method?

File: azkaban-common/src/main/java/azkaban/executor/ExecutorLoader.java
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorLoader.java:271]: comment about method functionality.

File: azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java
Comment (djaiswal83) [reply azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java:321]: Is this sleep added for debugging? I believe so.

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1127]: comment

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1129]: why not use StringBuffer ?

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1145]: Please use StringBuffer

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (djaiswal83) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1196]: Please use StringBuffer

File: azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java:321]: Yes, thank you.

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1081]: changed to status, ty.

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1080]: I believe deleteFlowContainer has a reference to the executionId but not the ExecutableFlow as it is reliant on the AzStatusMetadata.

File: azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java:81]: Added, thank you!

File: azkaban-common/src/main/java/azkaban/executor/ExecutorLoader.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/ExecutorLoader.java]: Added, thank you!

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1096]: Nit: Any reason for moving this inside the try block?

File: azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java
Comment (sshardool) [inline azkaban-common/src/main/java/azkaban/executor/JdbcExecutorLoader.java:311]: This is introducing a path where the webserver will update the flow-container logs stored in the db (possibly after the container is already gone).  This is not a problem per se, but does introduce a pattern of log updates we didn't have before. 
Would be good to keep the rest of the folks in the team in the loop for this as well as deploy and monitor the first fews clusters with this change more cautiously.

File: .travis.yml
Comment (sshardool) [reply .travis.yml:32]: Good to know, thanks for taking care of this.

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (sshardool) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1081]: Same for `deleteService`

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (sshardool) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java]: Yes, another point of the comment was that any delay added within `deletePod` method will automatically become a delay for the callers of deletePod. Some such callers, such as FlowStatusListenerManager are expected to be running mostly async tasks, possibly a very high number of them though and there processing could get delayed. 
We discussed this offline briefly, noting here for reference.

File: azkaban-common/src/main/java/azkaban/executor/container/ContainerizedImpl.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/container/ContainerizedImpl.java:22]: Sounds good!

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1132]: From my prior investigations of the logs it's pretty quick, though I haven't put a stopwatch on it, which is probably what is needed.

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1096]: A mistake :)

File: azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java
Comment (amielkelinkedin) [reply azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java:1083]: Done.

**Review by djaiswal83 — APPROVED** [review]
The changes look good to me. Please wait for @sshardool 's approval before merging the code.

**Review by sshardool — APPROVED** [review]
Thanks for the change. Approving based on the offline discussion including: https://github.com/azkaban/azkaban/pull/3027#discussion_r799165162

**Pull request merged**
